packages/engine/src/lib/services/variable-service.ts

```diff
@@ -86,11 +56,26 @@ export class VariableService {
     return value;
   }
 
+
+  private evalInScope(js: string, contextAsScope: Record<string, unknown>) {
+    try {
+      const keys = Object.keys(contextAsScope);
+      const values = Object.values(contextAsScope);
+      const functionBody = `return (${js})`;
+      const evaluatedFn = new Function(...keys, functionBody);
+      const result = evaluatedFn(...values);
+      return result ?? "";
+    } catch (exception) {
+      console.warn('Error evaluating expression', exception);
+      return "";
+    }
+  }
```

Comment:
Please note that this could be used to execute any JavaScript code, right? I mean this can resolve `{{process.env}}` for example or any other JavaScript code for that matter. I know this is executed inside `isolate` but for unsandboxed mode, this could be potentially harmful, I think we should work on restricting code that can be run next isA.

Reply:
Valid point, but let's revisit the isolation in the next sprint.

Running in isolated binary should be fine, as it's give the same privileged as js code.